packages/frontend/src/page/document_sidebar.tsx

```diff
@@ -0,0 +1,161 @@
+import { useNavigate } from "@solidjs/router";
+import type { Link } from "catlog-wasm";
+import { For } from "solid-js";
+import { Show } from "solid-js";
+import { createResource, createSignal } from "solid-js";
+import invariant from "tiny-invariant";
+
+import { type Api, type LiveDoc, useApi } from "../api";
+import { DocumentTypeIcon } from "../util/document_type_icon";
+import { DocumentMenu } from "./document_menu";
+import { AppMenu, ImportMenuItem, NewModelItem } from "./menubar";
+
+export function DocumentSidebar(props: {
+    liveDoc: LiveDoc;
+}) {
+    return (
+        <>
+            <AppMenu>
+                <NewModelItem />
+                <ImportMenuItem />
+            </AppMenu>
+            <RelatedDocuments liveDoc={props.liveDoc} />
+        </>
+    );
+}
+
+async function getLiveDocRoot(livDoc: LiveDoc, api: Api): Promise<LiveDoc> {
+    let parentLink: Link;
+    switch (livDoc.doc.type) {
+        case "diagram":
+            parentLink = livDoc.doc.diagramIn;
+            break;
+        case "analysis":
+            parentLink = livDoc.doc.analysisOf;
+            break;
+        default:
+            return livDoc;
+    }
+
+    const parentDoc = await api.getLiveDocFromLink(parentLink);
+    return getLiveDocRoot(parentDoc, api);
+}
+
+function RelatedDocuments(props: {
+    liveDoc: LiveDoc;
+}) {
+    const api = useApi();
+
+    // Signal to trigger refetch when documents are created
+    const [refreshCounter, setRefreshCounter] = createSignal(0);
+    const triggerRefresh = () => setRefreshCounter((c) => c + 1);
+
+    const [liveDocRoot] = createResource(
+        () => props.liveDoc,
+        async (liveDoc) => getLiveDocRoot(liveDoc, api),
+    );
+
+    return (
+        <Show when={liveDocRoot()} fallback={<div>Loading related items...</div>}>
+            {(liveDocRoot) => (
+                <div class="related-tree">
+                    <DocumentsTreeNode
+                        liveDoc={liveDocRoot()}
+                        indent={1}
+                        currentLiveDoc={props.liveDoc}
+                        refreshCounter={refreshCounter()}
+                        triggerRefresh={triggerRefresh}
+                    />
+                </div>
+            )}
+        </Show>
+    );
+}
+
+function DocumentsTreeNode(props: {
+    liveDoc: LiveDoc;
+    indent: number;
+    currentLiveDoc: LiveDoc;
+    refreshCounter: number;
+    triggerRefresh: () => void;
+}) {
+    const api = useApi();
+
+    const [childDocs] = createResource(
```

Comment:
I believe the more idiomatic way to manually cause a resource to refresh is to retain the refetch callback

```typescript
const [childDocs, { refetch }] = createResource(...)
```

and then call `refetch()` when needed. See: https://docs.solidjs.com/reference/basic-reactivity/create-resource